runtime/src/bank/partitioned_epoch_rewards/distribution.rs

```diff
@@ -129,6 +129,16 @@ impl Bank {
                 EpochRewardStatus::Active(EpochRewardPhase::Distribution(_))
             ));
             self.epoch_reward_status = EpochRewardStatus::Inactive;
+
+            self.epoch_rewards_calculation_cache
+                .lock()
+                .unwrap()
+                .remove(&self.epoch);
+            info!(
+                "removed epoch rewards calculation result for epoch {}",
+                self.epoch
+            );
```

Comment:
I think we can move the calculation cache entry for an epoch as soon as we root a block in the new epoch. But it's not necessarily true that when we are done distribution rewards that we've rooted any blocks in the new epoch yet. Can we change this to check for a new rooted block instead?